core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -3007,6 +3026,22 @@ private boolean createWalSymbolMapping(SymbolMapDiff symbolMapDiff, int columnIn
         return identical;
     }
 
+    private void dispatchO3CallbackQueue(RingQueue<O3CallbackTask> queue, int queuedCount) {
+        // This is work stealing, can run tasks from other table writers
+        final Sequence subSeq = this.messageBus.getO3CallbackSubSeq();
+        while (!o3DoneLatch.done(queuedCount)) {
+            long cursor = subSeq.next();
+            if (cursor > -1) {
+                O3CallbackJob.runCallbackWithCol(queue.get(cursor), cursor, subSeq);
+            }
+
+            if (cursor == -1) {
+                o3DoneLatch.await(queuedCount);
+            }
```

Comment:
nit: would be nice to add a backoff here similar to what we do in other such places:
```java
if (cursor == -2) {
    Os.pause();
}
```

Reply:
added